### ISSUE TITLE
Division / modulo not accurate.

### ISSUE THREAD

**Issue by orlp** [csaftoiu/GOLF-C#1]
Division and modulo in GOLF are not the same as they're in C, but you directly translate them.

**Comment by csaftoiu** [comment]
Noted, thanks. I'll update them when I get to fleshing out the code a bit more.

**Comment by orlp** [comment]
@csaftoiu Also keep in mind that bitshifts aren't the same either - not sure if you've implemented them yet.

**Comment by csaftoiu** [comment]
Yep I noticed that, I left them out for now. Actually a question about that: what's the difference between an arithmetic shift (sal, sar) and a logical shift (shl, shr)? It's not explained in the spec.

**Comment by orlp** [comment]
An arithmetic right shift shifts in the sign bit, logical shifts in zeroes. Normally arithmetic left shift == logical left shift, but because we have negative shift amounts they're different.

In C++ logical shift is on unsigned numbers while arithmetic shift is on signed numbers.
